### DISCUSSION TITLE
Select range of VLANs on interface

### DISCUSSION THREAD

**Discussion by rrheff** [netbox-community/netbox#8829] [Q&A]
How do you configure an interface's 802.1Q Mode with a range of allowed VLANs?

For example this Cisco swich interface configuration:
  switchport mode trunk
  switchport trunk allowed vlan 2-1001
  
I can choose Tagged or Tagged (All), but I don't see how to specify a range of VLANs without clicking each VLAN seperately. 
_Tagged: One untagged VLAN and/or one or more tagged VLANs
_Tagged (All): Implies all VLANs are available (w/optional untagged VLAN)__

**Comment by jeremystretch** [comment]
You need to create each of the VLANs in NetBox and assign them to the interface. A VLAN ID alone is not sufficient to uniquely identify a VLAN.

**Reply by rrheff** [reply]
That I can do. Is there an easy way of assigning a range of VLANs to an interface once they are created?
The GUI doesn't seem to support a comma separated list or something similar.

**Comment by mtinberg** [comment]
I don't think so, the data model really would prefer you associate the list of all relevant Netbox VLAN records to the interface specifically, which can be done over the API as well with a list of Netbox VLAN record ID numbers IIRC in tagged_vlans on the interface.

So in your case where you are pre-trunking 2-1001, you probably don't actually have 1000 active VLANs on that interface, you just aren't managing a curated list of what VLANs _are_ carried on that interface due to the additional management overhead. What I have done in my systems is add a custom field on interfaces called trunked_range​ and if there are more than 300 VLANs (higher than the spanning tree instance limit on our Catalyst devices) then I set the mode to trunked-all​ and fill in the trunked_range​ field for what should be in the (single) switchport trunk allowed vlan <trunked_range>​ line.  If there are less than 300 VLANs then I presume we are curating them manually and associating each with the Netbox VLAN record and building out the specific pruned list using Ansible jinja templates and their vlan_parser filter which breaks down a list of vlans into the right number of ranges and lines.



—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison
